core/src/main/java/org/elasticsearch/index/analysis/MultiTermAwareComponent.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.analysis;
+
+/** Elasticsearch counterpart of {@link org.apache.lucene.analysis.util.MultiTermAwareComponent}. */
+public interface MultiTermAwareComponent {
+
+    /** Returns an analysis component to handle analysis if multi-term queries.
+     * The returned component must be a TokenizerFactory, TokenFilterFactory or CharFilterFactory.
+     */
+    public Object getMultiTermComponent();
```

Comment:
Maybe it should have a default implementation that returns `this`? That seems like how you implement it most of the time.